tophat/opensource.tophat.com
# feat: Revamp the website with Jekyll

This PR updates the website to use Jekyll templates.

This addresses https://github.com/tophat/opensource.tophat.com/issues/64

# QA

- You need Ruby to [install Jekyll](https://jekyllrb.com/docs/). I had trouble installing Jekyll with the builtin system ruby so I installed RVM by following [this guide](https://usabilityetc.com/articles/ruby-on-mac-os-x-with-rvm/) but YMMV.
- Once Jekyll is installed, simply run the `bundle exec jekyll serve` command to get the site up and running.
- Verify the site looks as expected.
- Add or remove a project from the [projects.yaml](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_data/projects.yml) file (we can use JSON files instead of YAML files but I think YAML looks cleaner. More info [here](https://jekyllrb.com/docs/datafiles/)) and verify the projects show up as expected.

### Enabling Slack invite forms
- The Slack invite sections are also converted but they are disabled since Slack invite flow is broken at the moment. To enable the invite slack form at the bottom of the index page, uncomment [this](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_includes/join.html#L18-L24).
- To enable the slack invite page, remove `slack.html` from the [config.yml](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_config.yml#L5) file. Then restart the server (re-run `bundle exec jekyll serve`) for the changes to take effect.
